vault/token_store.go

```diff
@@ -1953,62 +1948,34 @@ func (ts *TokenStore) handleCreateCommon(ctx context.Context, req *logical.Reque
 		}
 		if role.Period != 0 {
 			switch {
-			case periodToUse == 0:
-				periodToUse = role.Period
+			case te.Period == 0:
+				te.Period = role.Period
 			default:
-				if role.Period < periodToUse {
-					periodToUse = role.Period
+				if role.Period < te.Period {
+					te.Period = role.Period
 				}
-				resp.AddWarning(fmt.Sprintf("Period specified both during creation call and in role; using the lesser value of %d seconds", int64(periodToUse.Seconds())))
+				resp.AddWarning(fmt.Sprintf("Period specified both during creation call and in role; using the lesser value of %d seconds", int64(te.Period.Seconds())))
 			}
 		}
 	}
 
 	sysView := ts.System()
 
-	if periodToUse > 0 {
-		// Cap period value to the sys/mount max value; this matches behavior
-		// in expiration manager for renewals
-		if periodToUse > sysView.MaxLeaseTTL() {
-			resp.AddWarning(fmt.Sprintf("Period of %d seconds is greater than current mount/system default of %d seconds, value will be truncated.", int64(periodToUse.Seconds()), int64(sysView.MaxLeaseTTL().Seconds())))
-			periodToUse = sysView.MaxLeaseTTL()
-		}
-		te.TTL = periodToUse
-	} else {
-		// Set the default lease if not provided, root tokens are exempt
-		if te.TTL == 0 && !strutil.StrListContains(te.Policies, "root") {
-			te.TTL = sysView.DefaultLeaseTTL()
+	// Only calculate a TTL if you are A) periodic, B) have a TTL, C) do not have a TTL and are not a root token
+	if te.Period > 0 || te.TTL > 0 || (te.TTL == 0 && !strutil.StrListContains(te.Policies, "root")) {
```

Comment:
See prior comment about encoding period/explicit max ttl in a token which is why the periodToUse stuff was here. Currently I don't think this is a good change; among other things it could be confusing to people as to why a token they look up says that the period is 60 minutes but it actually works like a token with a period set to 180 minutes.

Reply:
Ok, will add back periodToUse.